### ISSUE TITLE
Search syntax behaviour misleading (somehow)

### ISSUE THREAD

**Issue by ibressler** [SciCatProject/catanie#835]
The search field "Text Search" shows unexpected behavior: It seems to match full words only with logical 'OR' implicitly used for non-word characters such as hypens -  I'd expect it to find datasets with substring matching too.

Example 1: "20210906-46" matches other datasets sharing the date 20210906 only (found somewhere in a common 
path)

![Screenshot 2021-09-09 at 13 05 56](https://user-images.githubusercontent.com/638052/132675914-88ba26e2-5ea7-4dc1-b510-7d80e3dd0033.png)

Example 2: "2021090" Does not match anything.

![Screenshot 2021-09-09 at 13 06 27](https://user-images.githubusercontent.com/638052/132675956-39394c0e-8de1-4b8b-a934-d591f9d48bf4.png)

Same applies when searching for 'Creator information' partial names such as Owners, Investigators, ...

**Comment by stephan271** [comment]
Yes, unfortunately that is known "shortcoming" of the full text search as implemented inside Mongo, which only works on words, not on arbitrary substrings. Nothing that we can change. The only way out here , I think, is to create and fill an independent search engine for that (as was seemingly done by the Shanghai colleagues).
